bfss3/bfss3.go

```diff
@@ -57,10 +58,14 @@ import (
 // DefaultACL is the default ACL setting.
 const DefaultACL = "bucket-owner-full-control"
 
+var defaultHTTPClient = newHTTPClientWithoutCompression()
```

Comment:
I think this one should be public to be configurable by consumer, like `http.DefaultClient`

Reply:
I disagree, please revert.

Reply:
The right way to solve this is on `config.norm()`, we create the S3 client by:

```
client := s3.New(config.Session)
```

At this point `config.Session` (when not passed) should be based on http client with compression disabled